utils/determinants_tools.py

```diff
@@ -153,13 +156,37 @@ def csf_to_string(csf, bit_kind_size, size):
             import sys
             sys.exit(1)
 
-    with h5py.File(args.h5path, 'r') as f:
+    open_mode = 'r'
+    if args.transform:
+        open_mode = 'a'
+    with h5py.File(args.h5path, open_mode) as f:
+        alpha_group = 'MultiDet/CI_Alpha'
+        if alpha_group not in f:
+            alpha_group = 'MultiDet/CI_0'
+        beta_group = 'MultiDet/CI_Beta'
+        if beta_group not in f:
+            beta_group = 'MultiDet/CI_1'
 
-        bit_kind_size = f['MultiDet/CI_Alpha'].dtype.itemsize * 8
+        bit_kind_size = f[alpha_group].dtype.itemsize * 8
+        print(f[alpha_group].dtype)
 
         # We sanitize the determinants. Aka we transform any negative integers, into the proper unsigned one.
-        i_alpha= (sanitize(d,bit_kind_size) for d in f['MultiDet/CI_Alpha'])
-        i_beta= (sanitize(d,bit_kind_size) for d in f['MultiDet/CI_Beta'])
+        i_alpha= (sanitize(d,bit_kind_size) for d in f[alpha_group])
+        i_beta= (sanitize(d,bit_kind_size) for d in f[beta_group])
+        if args.transform:
+            alpha_dets = np.array([i for i in i_alpha], dtype=np.uint)
+            del f[alpha_group]
+            f.create_dataset(alpha_group, data=alpha_dets)
+            beta_dets = np.array([i for i in i_beta], dtype=np.uint)
+            del f[beta_group]
+            f.create_dataset(beta_group, data=beta_dets)
+            # regenerate iterators
+            i_alpha= (sanitize(d,bit_kind_size) for d in f[alpha_group])
+            i_beta= (sanitize(d,bit_kind_size) for d in f[beta_group])
+
+        if f[alpha_group].dtype == 'int64':
+            raise TypeError("This HDF5 file uses signed 64 bit integers (int64) to represent the determinants. QMCPACK expects unsigned integers (uint64) this can cause issues when used in qmcpack.")
```

Comment:
If we're not going to apply the signed->unsigned transformation here, then we should at least include the `--transform` flag in the error message.
I'd actually lean toward just removing the `--transform` option and doing the conversion if signed ints are detected.

Reply:
I actually like that I removed the transform option and always transform if we find signed ints